src/node_report.h

```diff
@@ -0,0 +1,186 @@
+#ifndef SRC_NODE_REPORT_H_
+#define SRC_NODE_REPORT_H_
+
+#include "v8.h"
+#include <uv.h>
+#include <node.h>
+#include <node_buffer.h>
+#include <algorithm>
+#include <cstring>
+#include <climits>
+#include <cstdlib>
+#include <utility>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <iostream>
+#include <sstream>
+#include <iomanip>
+
+#ifdef _WIN32
+#include <time.h>
+#else
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/time.h>
+#endif
+
+namespace nodereport {
+
+// enum DumpEvent {kException, kFatalError, kSignal_JS, kSignal_UV,
+// kJavaScript};
+
+#ifdef _WIN32
+typedef SYSTEMTIME TIME_TYPE;
+#else  // UNIX, OSX
+typedef struct tm TIME_TYPE;
+#endif
+
+typedef struct write_data {
+  std::ostream* out;
+  int* intend;
+} write_data;
+
+#define UNKNOWN_NODEVERSION_STRING "Unable to determine Node.js version\n"
+
+void InitializeNodeReport(v8::Isolate* isolate, node::Environment* env);
+
+// Function declarations - functions in src/node_report.cc
+void TriggerNodeReport(v8::Isolate* isolate, node::Environment* env,
+                       const char* message, const char* location,
+                       char* name, v8::MaybeLocal<v8::Value> error);
+void GetNodeReport(v8::Isolate* isolate,
+                   node::Environment* env,
+                   const char* message,
+                   const char* location,
+                   v8::MaybeLocal<v8::Value> error,
+                   std::ostream& out);
+
+// Function declarations - utility functions in src/utilities.cc
+void ReportEndpoints(uv_handle_t* h, std::ostringstream& out);
+void WalkHandle(uv_handle_t* h, void* arg);
+void WriteInteger(std::ostream& out, size_t value);
+#if defined(HAVE_OPENSSL)
+std::string GetOpenSSLVersion();
+#endif  // HAVE_OPENSSL
+
+
+// Function declarations - export functions in src/node_report_module.cc
+void TriggerReport(const v8::FunctionCallbackInfo<v8::Value>& info);
+void GetReport(const v8::FunctionCallbackInfo<v8::Value>& info);
+
+
+// Node.js boot time - defined in src/node_report.cc
+extern time_t boot_time;
+
+// JSON compiler definitions.
+#define ADVANCE(jstr, intend)                                                  \
+  do {                                                                         \
+    for (int i = 0; i < intend; i++) jstr << " ";                              \
+  } while (0)
+
+#define INTEND(marker) *marker += 2
```

Comment:
Isn't this supposed to be `INDENT` and `DEINDENT` or am I misunderstanding something?

Reply:
:) thanks. done. I guess I will never learn it properly.

Reply:
Oh, there seems to also be variables passed to this `define` (and as param names of other `define`s) named `intend` instead of `indent`.